core/bin/external_node/src/init.rs

```diff
@@ -78,25 +84,31 @@ pub(crate) async fn ensure_storage_initialized(
             .context("performing genesis failed")?;
         }
         InitDecision::SnapshotRecovery => {
-            anyhow::ensure!(
-                consider_snapshot_recovery,
+            let recovery_config = recovery_config.context(
                 "Snapshot recovery is required to proceed, but it is not enabled. Enable by setting \
                  `EN_SNAPSHOTS_RECOVERY_ENABLED=true` env variable to the node binary, or use a Postgres dump for recovery"
-            );
+            )?;
 
             tracing::warn!("Proceeding with snapshot recovery. This is an experimental feature; use at your own risk");
-            let recovery_config = SnapshotsRecoveryConfig::new()?;
-            let blob_store = ObjectStoreFactory::new(recovery_config.snapshots_object_store)
+            let object_store_config = snapshot_recovery_object_store_config()?;
+            let blob_store = ObjectStoreFactory::new(object_store_config)
                 .create_store()
                 .await;
 
             let config = SnapshotsApplierConfig::default();
-            let snapshots_applier_task = SnapshotsApplierTask::new(
+            let mut snapshots_applier_task = SnapshotsApplierTask::new(
                 config,
                 pool,
                 Box::new(main_node_client.for_component("snapshot_recovery")),
                 blob_store,
             );
+            if let Some(snapshot_l1_batch) = recovery_config.snapshot_l1_batch {
```

Comment:
"too old or non-existent" is not very helpful, we keep snapshots in gcs for 28 days, let's put info that you shouldn't use a snapshot older than 21 days

Reply:
That's an internal policy that can be changed at any point, and we'll probably forget to update the log here if it does.